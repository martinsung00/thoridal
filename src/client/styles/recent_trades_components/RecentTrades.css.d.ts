export const container: string;
export const root: string;
export const errorState: string;
export const trade: string;
export const content: string;
export const info: string;
export const note: string;
export const item: string;
export const comment: string;
export const additionalInfo: string;
export const details: string;
export const cost: string;
export const next: string;
export const nextGray: string;
export const prev: string;
export const prevGray: string;
export const controls: string;
export const controlUpdate: string;
export const controlDelete: string;
