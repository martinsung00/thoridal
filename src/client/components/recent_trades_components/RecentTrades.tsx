import React from "react";
import styles from "../../styles/recent_trades_components/RecentTrades.css";
import { Trade } from "../../types/index";

interface Props {
  trades: Array<Trade> | [];
  handleDelete: Function;
  history: Boolean;
  errored: Boolean;
}

export const RecentTrades: React.FunctionComponent<Props> = (props: Props) => {
  return (
    <div>
      <br />
      {props.errored && (
        <div>
          <div className={styles.errorState}>
            Uh Oh.. Something Has Gone Terribly Wrong! Try Again Later!
          </div>
        </div>
      )}
      {!props.history && !props.errored && (
        <div>
          <div className={styles.errorState}>
            Could Not Find Any Trade History
          </div>
        </div>
      )}
      {props.history && !props.errored && (
        <div className={styles.root}>
          <div style={{ marginTop: "-4em" }}>
            {(props.trades as Trade[]).map((trade: Trade, index: number) => {
              return (
                <div key={index} className={styles.trade}>
                  <div className={styles.content}>
                    <div className={styles.info}>
                      {" "}
                      <div>
                        Trade ID:
                        <div className={styles.item}>{trade.id}</div>
                      </div>
                      <br />
                      <div>
                        Reference Number:
                        <div className={styles.item}>
                          {trade.reference_number}
                        </div>
                      </div>
                      <br />
                      <div>
                        Ticker:
                        <div className={styles.item}>{trade.ticker}</div>
                      </div>
                      <br />
                      <div>
                        Company Name:
                        <div className={styles.item}>{trade.company_name}</div>
                      </div>
                      <br />
                      <br />
                    </div>
                    <div className={styles.note}>
                      <div className={styles.comment}>{trade.note}</div>
                      <br />
                      <div className={styles.additionalInfo}>
                        <div className={styles.details}>
                          {trade.trade_action === "bought" ? "Bought" : "Sold"}
                        </div>
                        <div className={styles.details}>
                          {trade.trade_type === "short" ? "Short" : "Long"}
                        </div>
                        <div className={styles.details}>
                          {trade.trade_status ? "Fulfilled" : "Not Fulfilled"}
                        </div>
                        <div className={styles.details}>
                          {`${trade.created_at}`.slice(0, 10)}
                        </div>
                        <br />
                        <div className={styles.cost}>
                          <div className={styles.details}>
                            Total Cost: ${trade.total_cost}
                          </div>
                          <div className={styles.details}>
                            Unit Price: ${trade.unit_price}
                          </div>
                          <div className={styles.details}>
                            Quantity: {trade.quantity}
                          </div>
                        </div>
                        <div className={styles.controls}>
                          <div
                            className={styles.controlUpdate}
                            id="update-button"
                          >
                            {"\u21ea"}
                          </div>
                          <div
                            className={styles.controlDelete}
                            id="delete-button"
                            onClick={function (
                              event: React.MouseEvent<
                                HTMLDivElement,
                                MouseEvent
                              >
                            ) {
                              const id = trade.id;
                              props.handleDelete(event, index, id);
                            }}
                          >
                            {"\u2297"}
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              );
            })}
          </div>
        </div>
      )}
    </div>
  );
};
