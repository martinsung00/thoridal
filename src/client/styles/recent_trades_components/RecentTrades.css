.container {
  position: absolute;
  width: 100em;
  left: 5em;
  margin-top: -2em;
  overflow: scroll;
}

.root {
  position: relative;
  margin-top: 15em;
  margin-left: 20em;
}

.errorState {
  position: relative;
  top: 6em;
  padding-bottom: 2em;
  text-align: center;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 40px;
}

.trade {
  position: relative;
  width: 72.15em;
  height: 14em;
  right: 16.1em;
  padding-top: 2em;
  border-bottom: 1.4px solid rgb(70, 70, 70);
}

.content {
  position: relative;
  font-family: Helvetica, Arial, sans-serif;
  left: 0;
  padding-top: 0.1em;
}

.info {
  position: relative;
  width: 7em;
  float: left;
}

.note {
  position: relative;
  width: 50em;
  font-size: 17px;
  float: right;
}

.item {
  position: relative;
  width: 10em;
  float: right;
  left: 10em;
  top: -1.2em;
  color: rgb(121, 121, 121);
  font-size: 17px;
  padding-bottom: 0.1em;
  padding-left: 0.05em;
}

.comment {
  position: relative;
  font-size: 17px;
  padding-bottom: 0.2em;
}

.additionalInfo {
  position: relative;
  margin-top: 0.1em;
  font-size: 13px;
  color: rgb(107, 107, 107);
}

.details {
  position: relative;
  float: left;
  padding-right: 1em;
}

.cost {
  position: relative;
  padding-top: 0.4em;
}

.next {
  position: relative;
  float: right;
  right: 22em;
  padding-top: 1.4em;
  padding-bottom: 1em;
  cursor: pointer;
  font-family: Helvetica, Arial, sans-serif;
}

.nextGray {
  position: relative;
  float: right;
  right: 22em;
  padding-top: 1.4em;
  padding-bottom: 1em;
  cursor: pointer;
  color: darkgray;
  font-family: Helvetica, Arial, sans-serif;
}

.next:hover {
  position: relative;
  float: right;
  right: 22em;
  padding-top: 1.4em;
  padding-bottom: 1em;
  cursor: pointer;
  color: darkgray;
  font-family: Helvetica, Arial, sans-serif;
}

.prev {
  position: relative;
  float: left;
  left: 2em;
  padding-top: 1.4em;
  padding-bottom: 1em;
  cursor: pointer;
  font-family: Helvetica, Arial, sans-serif;
}

.prevGray {
  position: relative;
  float: left;
  left: 2em;
  padding-top: 1.4em;
  padding-bottom: 1em;
  cursor: pointer;
  color: darkgray;
  font-family: Helvetica, Arial, sans-serif;
}

.prev:hover {
  position: relative;
  float: left;
  left: 2em;
  padding-top: 1.4em;
  padding-bottom: 1em;
  cursor: pointer;
  color: darkgray;
  font-family: Helvetica, Arial, sans-serif;
}

.controls {
  position: relative;
  padding-top: 1.4em;
  font-family: Helvetica, Arial, sans-serif;
}

.controlUpdate {
  position: relative;
  float: left;
  color: black;
  font-size: 14px;
  cursor: pointer;
}

.controlUpdate:hover {
  position: relative;
  float: left;
  color: rgb(8, 231, 0);
  font-size: 14px;
  cursor: pointer;
}

.controlDelete {
  position: relative;
  right: 36em;
  font-size: 22.2px;
  float: right;
  top: -0.13em;
  color: black;
  cursor: pointer;
}

.controlDelete:hover {
  position: relative;
  right: 36em;
  font-size: 22.2px;
  float: right;
  top: -0.13em;
  color: rgb(219, 0, 0);
  cursor: pointer;
}
