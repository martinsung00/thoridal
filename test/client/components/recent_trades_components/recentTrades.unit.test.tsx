import React from "react";
import renderer from "react-test-renderer";
import { RecentTrades } from "../../../../src/client/components/recent_trades_components/RecentTrades";
import { shallow, configure, ShallowWrapper } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Trade } from "../../../../src/client/types/index";

configure({ adapter: new Adapter() });

describe("Error Page", function () {
  const date = new Date("1995-12-17T03:24:00");
  const testTrades: Array<Trade> | [] = [
    {
      id: "TESTING",
      ticker: "TESTING",
      company_name: "TESTING",
      reference_number: "TESTING",
      unit_price: 1.0,
      quantity: 1,
      total_cost: 1.0,
      trade_type: "short",
      note: "TESTING",
      created_at: date,
      trade_status: false,
      trade_action: "bought",
    },
  ];

  const testTradesV2: Array<Trade> | [] = [
    {
      id: "TESTING",
      ticker: "TESTING",
      company_name: "TESTING",
      reference_number: "TESTING",
      unit_price: 1.0,
      quantity: 1,
      total_cost: 1.0,
      trade_type: "long",
      note: "TESTING",
      created_at: date,
      trade_status: true,
      trade_action: "sold",
    },
  ];

  const props = {
    trades: testTrades,
    handleDelete: jest.fn(),
    history: true,
    errored: false,
  };

  const propsV2 = {
    trades: testTradesV2,
    handleDelete: jest.fn(),
    history: true,
    errored: false,
  };

  const propsErrored = {
    trades: testTrades,
    handleDelete: jest.fn(),
    history: true,
    errored: true,
  };

  const propsNoHistory = {
    trades: [],
    handleDelete: jest.fn(),
    history: false,
    errored: false,
  };

  afterAll(function () {
    jest.resetAllMocks();
  });

  it("should render the page as is", function () {
    const container:
      | renderer.ReactTestRendererJSON
      | renderer.ReactTestRendererJSON[]
      | null = renderer.create(<RecentTrades {...props} />).toJSON();

    expect(container).toMatchSnapshot();
  });

  it("should render the trade's elements with 'long' and 'Fulfilled'", function () {
    const container:
      | renderer.ReactTestRendererJSON
      | renderer.ReactTestRendererJSON[]
      | null = renderer.create(<RecentTrades {...propsV2} />).toJSON();

    expect(container).toMatchSnapshot();
  });

  it("should render something has gone wrong text", function () {
    const wrapper: ShallowWrapper<
      any,
      Readonly<{}>,
      React.Component<{}, {}, any>
    > = shallow(<RecentTrades {...propsErrored} />);

    expect(wrapper).toMatchSnapshot();

    wrapper.unmount();
  });

  it("should render no trade history found text", function () {
    const wrapper: ShallowWrapper<
      any,
      Readonly<{}>,
      React.Component<{}, {}, any>
    > = shallow(<RecentTrades {...propsNoHistory} />);

    expect(wrapper).toMatchSnapshot();

    wrapper.unmount();
  });

  describe("delete button", function () {
    const flushPromises = () => new Promise(setImmediate);
    const event = {
      currentTarget: {},
    } as React.MouseEvent<HTMLInputElement>;

    beforeAll(function () {
      event.preventDefault = jest.fn();
    });

    afterAll(function () {
      jest.resetAllMocks();
    });

    it("should call handleDelete method", async function (done) {
      const wrapper: ShallowWrapper<
        any,
        Readonly<{}>,
        React.Component<{}, {}, any>
      > = shallow(<RecentTrades {...props} />);
      const spy = jest.spyOn<any, any>(props, "handleDelete");

      wrapper.find("#delete-button").simulate("click", event);
      await flushPromises();
      wrapper.update();

      expect(spy).toHaveBeenCalled();
      expect(wrapper).toMatchSnapshot();

      wrapper.unmount();

      done();
    });
  });
});
