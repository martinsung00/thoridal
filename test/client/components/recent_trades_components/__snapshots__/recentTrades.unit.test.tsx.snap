// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Page delete button should call handleDelete method 1`] = `ShallowWrapper {}`;

exports[`Error Page should render no trade history found text 1`] = `ShallowWrapper {}`;

exports[`Error Page should render something has gone wrong text 1`] = `ShallowWrapper {}`;

exports[`Error Page should render the page as is 1`] = `
<div>
  <br />
  <div>
    <div
      style={
        Object {
          "marginTop": "-4em",
        }
      }
    >
      <div>
        <div>
          <div>
             
            <div>
              Trade ID:
              <div>
                TESTING
              </div>
            </div>
            <br />
            <div>
              Reference Number:
              <div>
                TESTING
              </div>
            </div>
            <br />
            <div>
              Ticker:
              <div>
                TESTING
              </div>
            </div>
            <br />
            <div>
              Company Name:
              <div>
                TESTING
              </div>
            </div>
            <br />
            <br />
          </div>
          <div>
            <div>
              TESTING
            </div>
            <br />
            <div>
              <div>
                Bought
              </div>
              <div>
                Short
              </div>
              <div>
                Not Fulfilled
              </div>
              <div>
                Wed Feb 10
              </div>
              <br />
              <div>
                <div>
                  Total Cost: $
                  1
                </div>
                <div>
                  Unit Price: $
                  1
                </div>
                <div>
                  Quantity: 
                  1
                </div>
              </div>
              <div>
                <div
                  id="update-button"
                >
                  ⇪
                </div>
                <div
                  id="delete-button"
                  onClick={[Function]}
                >
                  ⊗
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Error Page should render the trade's elements with 'long' and 'Fulfilled' 1`] = `
<div>
  <br />
  <div>
    <div
      style={
        Object {
          "marginTop": "-4em",
        }
      }
    >
      <div>
        <div>
          <div>
             
            <div>
              Trade ID:
              <div>
                TESTING
              </div>
            </div>
            <br />
            <div>
              Reference Number:
              <div>
                TESTING
              </div>
            </div>
            <br />
            <div>
              Ticker:
              <div>
                TESTING
              </div>
            </div>
            <br />
            <div>
              Company Name:
              <div>
                TESTING
              </div>
            </div>
            <br />
            <br />
          </div>
          <div>
            <div>
              TESTING
            </div>
            <br />
            <div>
              <div>
                Sold
              </div>
              <div>
                Long
              </div>
              <div>
                Fulfilled
              </div>
              <div>
                Wed Feb 10
              </div>
              <br />
              <div>
                <div>
                  Total Cost: $
                  1
                </div>
                <div>
                  Unit Price: $
                  1
                </div>
                <div>
                  Quantity: 
                  1
                </div>
              </div>
              <div>
                <div
                  id="update-button"
                >
                  ⇪
                </div>
                <div
                  id="delete-button"
                  onClick={[Function]}
                >
                  ⊗
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
